### PR TITLE
[tests-only][full-ci] adding test for listing permissions of a project space after the space is shared to a user

### PR DESCRIPTION
**Pull request by ghost**
## Description

adding test for listing permissions of a project space after the space is shared to a user
```feature
Scenario: user lists permissions of a project space after sharing it
```

## Related Issue

- Related: https://github.com/owncloud/ocis/issues/8428 and https://github.com/owncloud/ocis/issues/8352

## How Has This Been Tested?

- test environment:
- locally
- ci
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 1]
```diff
@@ -92,33 +92,62 @@ public function createLinkShare(string $user, TableNode $body): ResponseInterfac
...
+	 *
+	 * @param string $user
+	 * @param string $fileOrFolder   (file|folder)
+	 * @param string $resource
+	 * @param string $space
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theUserGetsPermissionsListOfResource(string $user, string $fileOrFolder, string $resource, string $space):void {
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:133]:
```suggestion
	public function userGetsPermissionsListForResourceOfTheSpaceUsingTheGraphiAPI(string $user, string $fileOrFolder, string $resource, string $space):void {
```
Using the function name generated by behat will be good and unique for a function under the steps.

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 2]
```diff
@@ -92,33 +92,62 @@ public function createLinkShare(string $user, TableNode $body): ResponseInterfac
...
 	 * @param string $fileOrFolder   (file|folder)
 	 * @param string $resource
 	 * @param string $space
 	 *
-	 * @return void
+	 * @return ResponseInterface
 	 * @throws Exception
 	 */
-	public function theUserPermissionsListOfResource(string $user, string $fileOrFolder, string $resource, string $space):void {
+	public function theUserGetsPermissionsListOfResourceOrSpaceUsingTheGraphApi(string $user, string $fileOrFolder, string $resource, string $space):ResponseInterface {
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:103]:
```suggestion
	public function userGetsPermissionsListOfResourceOrSpace(string $user, string $fileOrFolder, string $resource, string $space):ResponseInterface {
```
For helper function don't use the text like `using ...`

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 3]
```diff
@@ -92,33 +92,62 @@ public function createLinkShare(string $user, TableNode $body): ResponseInterfac
...
+	/**
+	 * @When /^user "([^"]*)" lists the permissions of space "([^"]*)" using the Graph API$/
+	 *
+	 * @param string $user
+	 * @param string $space
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userGetsPermissionsListOfSpaceUsingTheGraphApi($user, $space):void {
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:148]:
```suggestion
	public function userListsThePermissionsOfSpaceUsingTheGraphApi($user, $space):void {
```
I think using the function name generated by behat will avoid the conflict.

File: tests/acceptance/expected-failures-localAPI-on-OCIS-storage.md [hunk 4]
```diff
@@ -262,6 +262,7 @@ The expected failures in this file are from features in the owncloud/ocis repo.
 - [apiSharingNg/deletePermissions.feature:141](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSharingNg/deletePermissions.feature#L141)
 - [apiSharingNg/deletePermissions.feature:161](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSharingNg/deletePermissions.feature#L161)
 - [apiSharingNg/deletePermissions.feature:179](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSharingNg/deletePermissions.feature#L179)
+- [apiSharingNg/listPermissions.feature:204](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiSharingNg/listPermissions.feature#L204)
```
Comment (saw-jan) [inline tests/acceptance/expected-failures-localAPI-on-OCIS-storage.md:265]:
```suggestion
```

**Review by saw-jan — CHANGES_REQUESTED** [review]
we should not add to the expected failure. also the issue is different

File: tests/acceptance/features/apiSharingNg/listPermissions.feature [hunk 5]
```diff
@@ -199,3 +199,205 @@ Feature: List a sharing permissions
...
       }
       """
+
+  @issues-8352
+  Scenario: user lists permissions of a project space after sharing it
+    Given using spaces DAV path
+    And user "Brian" has been created with default attributes and without skeleton files
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created a space "new-space" with the default quota using the Graph API
+    And user "Alice" has sent the following share invitation:
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/listPermissions.feature:209]: also create a link share of the space

File: tests/acceptance/features/apiSharingNg/listPermissions.feature [hunk 6]
```diff
@@ -199,3 +199,205 @@ Feature: List a sharing permissions
         }
       }
       """
+
+  @issues-8352
+  Scenario: user lists permissions of a project space after sharing it
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/listPermissions.feature:204]:
```suggestion
  Scenario: sharer lists permissions of a shared project space
```

File: tests/acceptance/features/apiSharingNg/listPermissions.feature [hunk 7]
```diff
@@ -209,116 +209,389 @@ Feature: List a sharing permissions
...
+                            "displayName": {
+                              "const": "Brian Murphy"
+                            }
+                          }
+                        }
+                      }
+                    },
+                    "id": {
+                      "type": "string",
+                      "pattern": "u:%user_id_pattern%$"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/listPermissions.feature:492]:
```suggestion
                      "pattern": "^u:%user_id_pattern%$"
```

**Review by saw-jan — APPROVED** [review]

**Pull request merged**